plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestModule.java

```diff
@@ -20,31 +20,11 @@
 package org.elasticsearch.plugin.ingest;
 
 import org.elasticsearch.common.inject.AbstractModule;
-import org.elasticsearch.common.inject.multibindings.MapBinder;
-import org.elasticsearch.ingest.processor.append.AppendProcessor;
-import org.elasticsearch.ingest.processor.convert.ConvertProcessor;
-import org.elasticsearch.ingest.processor.date.DateProcessor;
-import org.elasticsearch.ingest.processor.fail.FailProcessor;
-import org.elasticsearch.ingest.processor.geoip.GeoIpProcessor;
-import org.elasticsearch.ingest.processor.grok.GrokProcessor;
-import org.elasticsearch.ingest.processor.gsub.GsubProcessor;
-import org.elasticsearch.ingest.processor.join.JoinProcessor;
-import org.elasticsearch.ingest.processor.lowercase.LowercaseProcessor;
-import org.elasticsearch.ingest.processor.remove.RemoveProcessor;
-import org.elasticsearch.ingest.processor.rename.RenameProcessor;
-import org.elasticsearch.ingest.processor.set.SetProcessor;
-import org.elasticsearch.ingest.processor.split.SplitProcessor;
-import org.elasticsearch.ingest.processor.trim.TrimProcessor;
-import org.elasticsearch.ingest.processor.uppercase.UppercaseProcessor;
 import org.elasticsearch.plugin.ingest.rest.IngestRestFilter;
 
-import java.util.HashMap;
-import java.util.Map;
-
 public class IngestModule extends AbstractModule {
 
     private final boolean ingestEnabled;
-    private final Map<String, ProcessorFactoryProvider> processorFactoryProviders = new HashMap<>();
 
     public IngestModule(boolean ingestEnabled) {
         this.ingestEnabled = ingestEnabled;
```

Comment:
All the transport actions and components that really do something are in the plugin. The reason why this flag is here is to disabled / not load all this stuff in the plugin when the plugin is disabled, which right now is the default. Core only holds interfaces, base classes and a place to register plugins. Whether ingest is going to be disabled or enabled by default is still undecided.